### PR TITLE
RFC: slack support

### PR DESCRIPTION
**Pull request by ktdreyer**
This is my attempt at a native Slack support. Slack provides a "Real Time Messaging" API via Websockets, so I've used Autobahn for async handling in Twisted's framework.

I've tested this for a couple weeks in my personal Slack.com org and Helga can respond in channels, send users direct IMs, etc.

Slack does have IRC and XMPP connectors, but a native Slack backend for Helga will mean that Helga can use the "bot" API tokens, post richly-formatted responses in Slack, and in general, have full feature parity with other Slack-aware chat bots.

There are several things left to do here, ranging from "more docs" (my current doc changes are pitiful) to "more async" (use treq instead of requests) to "more features", but I wanted to get your initial feedback, and in particular, what items are most important to you before merging to master.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -35,6 +35,9 @@
 if sys.version_info[:2] == (2, 6):
     extra_requires = ['argparse==1.3.0']
 
+if sys.version_info[1:2] < (3, 3):
+    extra_requires = ['backports.functools_lru_cache']
+
```
Comment (ktdreyer) [inline setup.py]: bah, I forgot I was using lru_cache in a previous version of this change. But I'm not using it any more (it's not flexible enough for my needs.) I will drop it as part of the future changes.

**Comment by shaunduncan** [comment]
This looks great. I thought about documenting a way to do this using a combination of the IRC gateway and a webhook plugin, but never made any progress. I think this will be a lot more flexible for standard slack integrations. From a quick glance it looks pretty good to me. I may try to play around with this a bit and wrap this into a 1.8 release if that works.

**Comment by shaunduncan** [comment]
@ktdreyer are you ok with a bit of a delay here for a 1.8 release? Just wanted to make sure you weren't super pressed for support for this.

**Comment by ktdreyer** [comment]
This can definitely wait, and it's more important to me that the Twisted unpinning change be pushed to PyPI.

Thanks for looking it over!

**Comment by shaunduncan** [comment]
Sounds good. I'll push up a 1.7.4 with the recent merges.

**Comment by shaunduncan** [comment]
Some issues with PyPI, but it's actually 1.7.5: https://pypi.python.org/pypi/helga/1.7.5

**Comment by narfman0** [comment]
@ktdreyer twisted has been unpinned, and some unit test stuff has been updated. You might want to rebase/update this branch (and subsequently rerun tests) :)

**Comment by shaunduncan** [comment]
I'm definitely going to need to pull this in and experiment a bit with it now that slack is deprecating the IRC gateway

**Comment by ktdreyer** [comment]
Please edit at will - I have not had time to look into this for a while :)

The optional JSON-parsing stuff in `msg()` always struck me as a little weird. The idea was that plugins would be able to post rich Slack messages to channels. Maybe it would be better to have plugins return a custom `SlackMessage(str)` class with a `message_data` attribute or something.

**Comment by shaunduncan** [comment]
@ktdreyer merging this now and continuing with a bit of experimentation and any slight updates (if at all needed), thank you so much for this!

**Pull request merged**